### PR TITLE
Fixed boolean type not working with char(1) and bit(1)

### PR DESCRIPTION
**Pull request by yongzhi-chen**
### Description

This patch fixes boolean type not working correctly. Previous code fails on test case where data type is CHAR(0) and BIT(0)

@bajtos 

#### Related issues

- connect to #325

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html) 
(The codebase of this repo is using old javascript style so most of the style guide doesnt really apply...)

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by yongzhi-chen** [comment]
@bajtos not sure what commit linter is about. The link goes to 404.

**Comment by dhmlau** [comment]
@yongzhi-chen, thanks for your PR.  The commit linter error says:
```
**  Linting commit logs
**
**  1 problems found:
**    bd1b32b - Fixed boolean type not working with char(1) and bit: First line should be 50 characters or less (saw 54)
**
```

You can fix the commit message by following the instructions: https://help.github.com/en/articles/changing-a-commit-message. 
Please also see our commit message guideline for your future reference: https://loopback.io/doc/en/contrib/git-commit-messages.html

**Comment by yongzhi-chen** [comment]
@dhmlau Thanks I changed commit message. let me know if there are other issues. CI build error looks like it is configuration problem.

**Comment by dhmlau** [comment]
@yongzhi-chen, I've rebased your branch with master. Let's see if CI passes.

File: lib/mysql.js [hunk 1]
```diff
@@ -435,7 +435,10 @@ MySQL.prototype.fromColumnValue = function(prop, val) {
         }
         break;
       case 'Boolean':
-        val = Boolean(val);
+        // BIT(1) case: <Buffer 01> for true and <Buffer 00> for false
+        // CHAR(1) case: '1' for true and '0' for false
+        // TINYINT(1) case: 1 for true and 0 for false
+        val = Buffer.isBuffer(val) ? Boolean(val[0]) : Boolean(parseInt(val));
```
Comment (bajtos) [inline lib/mysql.js]: Should we act more defensively and check that the buffer has a single byte only, before treating it as a `BIT(1)`?

```suggestion
        val = Buffer.isBuffer(val) && val.length === 1 ? Boolean(val[0]) : Boolean(parseInt(val));
```

**Comment by dhmlau** [comment]
We just switch the contribution method from CLA to DCO, making your contribution easier in the future. Please sign the commits with DCO by amending your commit messages with `-s` flag and push the changes again. If you're using the command line, you can:

```
git commit --amend -s
git push --force-with-lease
```

Please refer to [this docs page](https://loopback.io/doc/en/contrib/code-contrib.html#developer-certificate-of-origin-dco) for details. Thanks! 

For more questions, please join our [Slack workspace - #loopback-contributors](https://loopbackio.slack.com/join/shared_invite/zt-8lbow73r-SKAKz61Vdao~_rGf91pcsw#/).

**Comment by dhmlau** [comment]
Closing in favor of #446

**Pull request closed without merging**